src/utils.js

```diff
@@ -166,10 +166,24 @@ export function generateName() {
   return name.replace(/^./, name[0].toUpperCase());
 }
 
-export function promptForName() {
-  var username = generateName();
+export function promptForName(username) {
+  if (!username)
+    username = generateName();
+  
   do {
     username = prompt("Choose a username", username);
   } while (!(username && username.length));
   return username;
 }
+
+export function getCookie(name) {
+  var value = "; " + document.cookie;
+  var parts = value.split("; " + name + "=");
+  if (parts.length == 2) return parts.pop().split(";").shift();
+}
+
+export function parseJwt (token) {
+  var base64Url = token.split('.')[1];
+  var base64 = base64Url.replace('-', '+').replace('_', '/');
+  return JSON.parse(window.atob(base64));
+}
```

Comment:
Oh. Is the token not signed and encrypted?

Reply:
If not, maybe the thing creating the cookie can just make it a json string to begin with so we don't have to do this little dance here.

Reply:
JWT are signed, but not encrypted